.github/workflows/deploy-dev.yml

```diff
@@ -0,0 +1,90 @@
+name: Deploy Dev Enviroment
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'env-dev.yml'
+  # TODO: Figure out an automatic trigger. Using manual trigger for now
+  workflow_dispatch:
+env:
+    NAME: payu-dev
+    VERSION: dev
+jobs:
+  pack:
+    name: Pack Payu
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Micromamba
+        uses: mamba-org/setup-micromamba@v1.7.0
+        with:
+          micromamba-version: '1.5.3-0'
+          environment-file: env.yml
+          environment-name: ${{ env.NAME }}
+          generate-run-shell: true
+
+      - name: Create Pack and Lockfile
+        shell: micromamba-shell {0}
+        run: |
+          conda pack
+          conda-lock lock --file env.yml --platform linux-64 --micromamba --lockfile ${{ env.NAME }}.conda-lock.yml
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ env.NAME }}
+          if-no-files-found: error
+          path: |
+            ${{ env.NAME }}.tar.gz
+            ${{ env.NAME }}.conda-lock.yml
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs:
+      - pack
+    environment: Gadi
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/download-artifact@v3.0.2
+        with:
+          name: ${{ env.NAME }}
+
+      - uses: access-nri/actions/.github/actions/setup-ssh@main
+        id: ssh
+        with:
+          hosts: |
+            ${{ secrets.HOST_DATA }}
+            ${{ secrets.HOST }}
+          private-key: ${{ secrets.SSH_KEY }}
+
+      - name: Copy to Gadi 
+        run: |
+          rsync -e 'ssh -i ${{ steps.ssh.outputs.private-key-path }}' \
+            ${{ env.NAME }}.tar.gz \
+            ${{ secrets.USER }}@${{ secrets.HOST_DATA }}:${{ vars.PACK_LOCATION }}
+
+      - name: Deploy to Gadi
+        env:
+          PAYU_ENVIRONMENT_LOCATION: ${{ vars.PRERELEASE_DEPLOYMENT_LOCATION }}/${{ env.VERSION }}
+        run: |
+          ssh ${{ secrets.USER }}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
+          
+          # Remove previous enviroment if it exists
+          if [ -d ${{ env.PAYU_ENVIRONMENT_LOCATION }} ]; then
+            rm -rf ${{ env.PAYU_ENVIRONMENT_LOCATION }}
+          fi
```

Comment:
I'm wondering what effect this will have on someone who is currently using this environment?

We could install into a location that is uniquely named (date and git hash?) and then change a symlink to point to this once it is installed. That way anyone currently using the previous latest env will be unaffected, but pick up a new env when they load the module the next time.

So if we kept newest and previous it wouldn't be a big burden (resource wise). It would require a bit of CI to remove the previous environment when we update.

Reply:
Yeah that's could a be a problem.. If we used date & git hash, would it be the git hash of payu repository? 
So have a `/g/data/prerelease/apps/payu/dev` symlink, deploy to `/g/data/prerelease/apps/payu/dev-11-04-23_91f2831` (or something like that?) . When updating the `/g/data/prerelease/apps/payu/dev` symlink, if a previously environment was loaded, `module unload` would run activate script of the new environment. But the environment variables that are set change might stay the same (as use the symlink path), for example the env vars set by payu 1.1 are:
```
jb4202@gadi-login-04 ~ $ module show payu
-------------------------------------------------------------------
/g/data/vk83/modules/payu/1.1:

module-whatis   {Activate Payu's micromamba environment}
conflict        payu
setenv          GSETTINGS_SCHEMA_DIR_CONDA_BACKUP {}
setenv          CONDA_PREFIX /g/data/vk83/apps/payu/1.1
setenv          GSETTINGS_SCHEMA_DIR /g/data/vk83/apps/payu/1.1/share/glib-2.0/schemas
setenv          _CONDA_PACK_OLD_PS1 {}
setenv          SHLVL 1
prepend-path    PATH /g/data/vk83/apps/payu/1.1/bin
setenv          XML_CATALOG_FILES {file:///g/data/vk83/apps/payu/1.1/etc/xml/catalog file:///etc/xml/catalog}
setenv          LC_ALL en_AU.utf8
```

So when the symlink changes, all of sudden using a different directory for the environment which might also not be ideal - but  is that better than the directory no longer existing..

Reply:
Resolving this as now installing to location that is named with date and git hash, using symlink to point to new environment, and keeping newest and previous